src/CattyTests/Extensions/NSStringExtensionTests.swift

```diff
@@ -0,0 +1,253 @@
+
+        let testString3 = NSString(string: "a")
+        let fromString3 = "a"
+        let toString3 = "a"
+        let correctOutput3: String? = nil
+
+        let testString4 = NSString(string: "")
+        let fromString4 = ""
+        let toString4 = ""
+        let correctOutput4: String? = nil
+
+        let testString5 = NSString(string: "abcdefg")
+        let fromString5 = "xy"
+        let toString5 = "pq"
+        let correctOutput5: String? = nil
+
+        let testString6 = NSString(string: "abcdefg")
+        let fromString6 = "ef"
+        let toString6 = "ab"
+        let correctOutput6: String? = nil
+
+        let testString7 = NSString(string: "abcdefabxyef")
+        let fromString7 = "ab"
+        let toString7 = "ef"
+        let correctOutput7 = "cd"
+
+        let testString8 = NSString(string: "<abcxyzabc")
+        let fromString8 = "abc"
+        let toString8 = "abc"
+        let correctOutput8 = "xyz"
+
+        XCTAssertEqual(testString1.stringBetweenString(fromString1, andString: toString1, withOptions: .caseInsensitive), correctOutput1)
+        XCTAssertEqual(testString2.stringBetweenString(fromString2, andString: toString2, withOptions: .caseInsensitive), correctOutput2)
+        XCTAssertEqual(testString3.stringBetweenString(fromString3, andString: toString3, withOptions: .caseInsensitive), correctOutput3)
+        XCTAssertEqual(testString4.stringBetweenString(fromString4, andString: toString4, withOptions: .caseInsensitive), correctOutput4)
+        XCTAssertEqual(testString5.stringBetweenString(fromString5, andString: toString5, withOptions: .caseInsensitive), correctOutput5)
+        XCTAssertEqual(testString6.stringBetweenString(fromString6, andString: toString6, withOptions: .caseInsensitive), correctOutput6)
+        XCTAssertEqual(testString7.stringBetweenString(fromString7, andString: toString7, withOptions: .caseInsensitive), correctOutput7)
+        XCTAssertEqual(testString8.stringBetweenString(fromString8, andString: toString8, withOptions: .caseInsensitive), correctOutput8)
+
+    }
+
+    func testIsValidNumber() {
+
+        let testString1 = NSString(string: "1")
+        let testString2 = NSString(string: "1.2")
+        let testString3 = NSString(string: "11.22")
+        let testString4 = NSString(string: "11.12.14")
+        let testString5 = NSString(string: "1,2")
+        let testString6 = NSString(string: "a")
+        let testString7 = NSString(string: "1a")
+        let testString8 = NSString(string: "1.a")
+
+        XCTAssertTrue(testString1.isValidNumber())
+        XCTAssertTrue(testString2.isValidNumber())
+        XCTAssertTrue(testString3.isValidNumber())
+        XCTAssertFalse(testString4.isValidNumber())
+        XCTAssertFalse(testString5.isValidNumber())
+        XCTAssertFalse(testString6.isValidNumber())
+        XCTAssertFalse(testString7.isValidNumber())
+        XCTAssertFalse(testString8.isValidNumber())
+
+    }
+
+    func testContainsString() {
```

Comment:
You can remove this test since this is a standard Swift String functionality. However, please add tests for `uuid`.